pkg/cmd/operator/webhooks.go

```diff
@@ -138,25 +142,37 @@ func (o *WebhookOptions) Complete() error {
 			return err
 		}
 
-		o.servingCertificate.Certificate = [][]byte{derBytes}
-
+		o.TLSConfig = &tls.Config{
+			Certificates: []tls.Certificate{
+				{
+					Certificate: [][]byte{derBytes},
+					PrivateKey:  privateKey,
+				},
+			},
+		}
 	} else {
-		o.servingCertificate, err = tls.LoadX509KeyPair(o.TLSCertFile, o.TLSKeyFile)
+		o.dynamicCertKeyPairContent, err = dynamiccertificates.NewDynamicServingContentFromFiles("serving-certs", o.TLSCertFile, o.TLSKeyFile)
 		if err != nil {
-			return err
+			return fmt.Errorf("can't create DynamicServingContentFromFiles: %w", err)
+		}
+
+		o.TLSConfig = &tls.Config{
+			GetCertificate: func(*tls.ClientHelloInfo) (*tls.Certificate, error) {
+				cert, err := tls.X509KeyPair(o.dynamicCertKeyPairContent.CurrentCertKeyContent())
+				return &cert, err
+			},
 		}
 	}
 
+	o.assignedPortCh = make(chan struct{})
```

Comment:
I'm not a fan of this. It's supposed to be internal and used for testing and we don't use the constructor in tests. So it would either require using the constructor in tests, validating it or explicitly creating the channel in each struct in tests?